### PR TITLE
Allowed rematches to occur if I_VS_SEEKER_CHARGING isn't enabled

### PR DESCRIPTION
**Pull request by fdeblasio**
## Description
The way the code is currently written, the step counter for rematches will never increment if `I_VS_SEEKER_CHARGING` is disabled

## **Discord contact info**
Frankfurter

### CODE REVIEW

**Comment by pkmnsnfrn** [comment]
In plain English, does this mean that:

On `master` right now, if the Match Call is enabled _and_ the player does not have the Vs. Seeker, they will never get rematches because the counter does not increment?

**Comment by fdeblasio** [comment]
> In plain English, does this mean that:
> 
> On `master` right now, if the Match Call is enabled _and_ the player does not have the Vs. Seeker, they will never get rematches because the counter does not increment?

Correct

File: src/battle_setup.c [hunk 1]
```diff
@@ -1939,8 +1939,10 @@ void IncrementRematchStepCounter(void)
 {
 #if FREE_MATCH_CALL == FALSE
     if (HasAtLeastFiveBadges()
-        && (I_VS_SEEKER_CHARGING != 0)
-        && (!CheckBagHasItem(ITEM_VS_SEEKER, 1)))
+#if I_VS_SEEKER_CHARGING != 0
```
Comment (pkmnsnfrn) [inline src/battle_setup.c:1942]:
```c
void IncrementRematchStepCounter(void)
{
#if FREE_MATCH_CALL == FALSE
    if (!HasAtLeastFiveBadges())
        return;
    
    if (IsVsSeekerEnabled())
        return;

    if (gSaveBlock1Ptr->trainerRematchStepCounter >= STEP_COUNTER_MAX)
        gSaveBlock1Ptr->trainerRematchStepCounter = STEP_COUNTER_MAX;
    else
        gSaveBlock1Ptr->trainerRematchStepCounter++;
#endif //FREE_MATCH_CALL
}
```

I think this is easier to parse.

File: src/battle_setup.c [hunk 2]
```diff
@@ -1939,8 +1939,10 @@ void IncrementRematchStepCounter(void)
...
+        && (!CheckBagHasItem(ITEM_VS_SEEKER, 1))
```
Comment (pkmnsnfrn) [inline src/battle_setup.c:1943]:
```c
bool32 IsVsSeekerEnabled(void);
// add to include/vs.seeker.h
```


```c
bool32 IsVsSeekerEnabled(void)
{
    if (I_VS_SEEKER_CHARGING == 0)
        return FALSE;
    
    return (CheckBagHasItem(ITEM_VS_SEEKER, 1));
}
// add to src/vs.seeker.c
```

**Review by pkmnsnfrn — CHANGES_REQUESTED** [review]
Okay this is my first review so I truly have no idea what I'm doing. I tried to do this in codespace but it didn't quite work, so hopefully this does.

File: src/battle_setup.c (hunk 1 above)
Comment (fdeblasio) [reply src/battle_setup.c]: Done!

File: src/battle_setup.c (hunk 2 above)
Comment (fdeblasio) [reply src/battle_setup.c]: Done!

**Review by pkmnsnfrn — DISMISSED** [review]

**Comment by pkmnsnfrn** [comment]
I have tested the following scenarios:

- When the player has more than four badges, and 
    - `I_VS_SEEKER_CHARGING` is not set, `IncrementRematchStepCounter` runs and `UpdateVsSeekerStepCounter` does not.
    - `I_VS_SEEKER_CHARGING` is  not set, and the player has `ITEM_VS_SEEKER`, `IncrementRematchStepCounter` runs and `UpdateVsSeekerStepCounter` does not.
    - `I_VS_SEEKER_CHARGING` is set, and the player has `ITEM_VS_SEEKER`, `IncrementRematchStepCounter` does not run and `UpdateVsSeekerStepCounter` runs.
- When the player has less than five badges, neither function runs.

**Comment by AlexOn1ine** [comment]
@fdeblasio could you please retarget this to master? It is an improtant bugfix that I think 1.8.5 should get.

**Comment by fdeblasio** [comment]
> @fdeblasio could you please retarget this to master? It is an improtant bugfix that I think 1.8.5 should get.

Done! I think it needs re-approval though

**Comment by AlexOn1ine** [comment]
> > @fdeblasio could you please retarget this to master? It is an improtant bugfix that I think 1.8.5 should get.
> 
> Done! I think it needs re-approval though

thx

**Review by AlexOn1ine — APPROVED** [review]

**Comment by AlexOn1ine** [comment]
@fdeblasio weird ask but could you squash the pr and force push so CI runs again?

**Comment by fdeblasio** [comment]
> @fdeblasio weird ask but could you squash the pr and force push so CI runs again?

Done!

**Comment by AlexOn1ine** [comment]
> > @fdeblasio weird ask but could you squash the pr and force push so CI runs again?
> 
> Done!

Thx

**Review by AlexOn1ine — APPROVED** [review]
